src/integer/z/sample.rs

```diff
@@ -0,0 +1,147 @@
+        let upper_bound = Z::from(32);
+        for _i in 0..32 {
+            let sample = Z::sample_uniform(&lower_bound, &upper_bound).unwrap();
+            assert!(lower_bound <= sample);
+            assert!(sample < upper_bound);
+        }
+    }
+
+    /// Checks whether the boundaries of the interval are kept for large intervals.
+    #[test]
+    fn boundaries_kept_large() {
+        let lower_bound = Z::from(i64::MIN) - Z::from(u64::MAX);
+        let upper_bound = Z::from(i64::MIN);
+        for _i in 0..256 {
+            let sample = Z::sample_uniform(&lower_bound, &upper_bound).unwrap();
+            assert!(lower_bound <= sample);
+            assert!(sample < upper_bound);
+        }
+    }
+
+    /// Checks whether providing an invalid interval results in an error.
+    #[test]
+    fn invalid_interval() {
+        let lb_0 = Z::from(i64::MIN) - Z::ONE;
+        let lb_1 = Z::from(i64::MIN);
+        let lb_2 = Z::ZERO;
+        let upper_bound = Z::from(i64::MIN);
+
+        let res_0 = Z::sample_uniform(&lb_0, &upper_bound);
+        let res_1 = Z::sample_uniform(&lb_1, &upper_bound);
+        let res_2 = Z::sample_uniform(&lb_2, &upper_bound);
+
+        assert!(res_0.is_err());
+        assert!(res_1.is_err());
+        assert!(res_2.is_err());
+    }
+
+    /// Checks whether `sample_uniform` is available for all types
+    /// implementing Into<Z> + Clone, i.e. u8, u16, u32, u64, i8, ...
+    #[test]
+    fn availability() {
+        let modulus = Modulus::from_str("7").unwrap();
+        let zq = Zq::from_str("7 mod 10").unwrap();
+        let z = Z::from(7);
+
+        let _ = Z::sample_uniform(&0u16, &7u8);
+        let _ = Z::sample_uniform(&0u32, &7u16);
+        let _ = Z::sample_uniform(&0u64, &7u32);
+        let _ = Z::sample_uniform(&0i8, &7u64);
+        let _ = Z::sample_uniform(&0i16, &7i8);
+        let _ = Z::sample_uniform(&0i32, &7i16);
+        let _ = Z::sample_uniform(&0i64, &7i32);
+        let _ = Z::sample_uniform(&Z::ZERO, &7i64);
+        let _ = Z::sample_uniform(&0u8, &modulus);
+        let _ = Z::sample_uniform(&0, &zq);
+        let _ = Z::sample_uniform(&0, &z);
+    }
+
+    /// Roughly checks the uniformity of the distribution.
+    /// This test could possibly fail for a truly uniform distribution
+    /// with probability smaller than 1/1000.
+    #[test]
+    fn uniformity() {
+        let lower_bound = Z::ZERO;
+        let upper_bound = Z::from(5);
+        let mut counts = [0; 5];
+        // count sampled instances
+        for _i in 0..1000 {
+            let sample_z = Z::sample_uniform(&lower_bound, &upper_bound).unwrap();
+            let sample_int = i64::try_from(&sample_z).unwrap() as usize;
+            counts[sample_int] += 1;
+        }
+
+        // Check that every sampled integer was sampled roughly the same time
+        // this could possibly fail for true uniform randomness with probability
+        for i in 0..5 {
+            assert!(counts[i] > 150);
+        }
```

Comment:
I had the same concerns and let this test run ~18.000 times. From these 18.000 tries, only 3 failed, i.e. the real probability of this test to fail is should probably be less than 1/5000. I just didn't want to explicitely calculate the probability for this to fail and therefore estimated the probability conservatively to the next 10^k step.

As I only include this test for `Z` and `Zq` to ensure that `Zq`s reduction does not hide a number that was sampled more often (i.e. all instances that are too large, get thrown out), we will probably only have up to 10 of these tests in the math crate and the bounds need to stay roughly in this position to really check the validity that no number was sampled more often than another one.

Hence, I hope that with your suggestion regarding assertion explanation included, you do not worry about the pipeline to fail too much and otherwise - with very low probability - we have to check the CIs output and re-run the pipeline.